pycfmodel/model/resources/properties/policy_document.py

```diff
@@ -174,12 +174,12 @@ def allowed_actions_with(self, pattern: Pattern) -> List[Statement]:
             if statement.actions_with(pattern) and statement.Effect == "Allow"
         ]
 
-    def allowed_principals_with(self, pattern: Pattern) -> List[Statement]:
+    def allowed_principals_with(self, pattern: Pattern) -> Iterator[Statement]:
```

Comment:
It'd be nice to replace the generators by actual lists. CF files are not big enough for a generator to be needed and makes dealing with this easier.